Update: add details to GraphQL docs (#3938)
aws-amplify/docs: src/pages/cli/graphql/client-code-generation.mdx

> When you deploy your GraphQL API to the cloud, you are prompted to configure codegen. When a project is configured to generate code with codegen, it stores all the configuration `.graphqlconfig.yml` file in the root folder of your project.

Reviewer: We should explicitly state that customers should not modify this file if we're calling it out.

Author: Updated

> Models are not automatically generated by codegen on an `amplify push`.

Reviewer: Why aren't we triggering modelgen? If it's necessary for their app and they've configured it upon prompt then we should do it automatically.

Author: It doesn't seem like anyone has clear information on this after investigating, so I'm removing this and there will be a follow up discussion/change

> Models will not be automatically generated by codegen on an `amplify push`.

Reviewer 2: ```suggestion
Models are not automatically generated by codegen on an `amplify push`.
```
